verifier/src/lib.rs

```diff
@@ -34,15 +34,31 @@ pub enum VerificationError {
 
 /// A verifier. These can chained together using the [`Or`] and [`And`]
 /// types.
-pub trait Verifier: Debug {
+pub trait Verifier<T>: Debug {
     /// The error that this verification will return in failure cases.
     type Error;
 
-    /// Performs a verification operation for the [`Verifier`].
+    /// Performs a verification operation on `evidence`.
     ///
     /// In order to accommodate constant time operations this returns a
     /// [`CtOption`] instead of a [`Result`].
-    fn verify(&self) -> CtOption<Self::Error>;
+    fn verify(&self, evidence: &T) -> CtOption<Self::Error>;
+
+    /// Or this verifier with another.
+    fn or<U, V: Verifier<U>>(self, other: V) -> Or<Self, V>
```

Comment:
Up for thoughts here I thought it would be nicer to `verifier.and(other)` instead of having to `And::new(verifier, other)` all the time.

Reply:
Thinking some more this might be harder to understand when one builds the tree, i.e.
```rust
a.and(b).and(c)
```
This will result in the following tree, which may not be as intuitive.
```
           And
           /\
          /  \
         /    \
       And     C
       /\
      /  \
     /    \
    A      B
```

Reply:
Back in the day, GCC and Clang would turn `a && b && c` into either `a && (b && c)` or `(a && b) && c` based on RNG by default, so I don't think this is a huge deal ;-)

Reply:
Wow, I thought the C spec specified it should be (a && b) && c